Subarray-Equals-k.java

```diff
@@ -0,0 +1,22 @@
+// Time Complexity : O(N)
+// Space Complexity :O(N)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : A lot of thinking to process this approach. Confused 
```

Comment:
Please try to extend Contiguous array problem. If still confused, Please DM me in slack.